src/Assets/LightSkyDefense/Scripts/Enemy.cs

```diff
@@ -241,33 +199,59 @@ private void OnTriggerEnter(Collider collider)
             return;
         }
 
-        var pathPoints = GameManager.Instance.Path.PathPoints;
+        var foundIndex = int.Parse(collider.gameObject.name.Substring(3));
 
-        _waypointIndex = Mathf.Clamp(
-            int.Parse(collider.gameObject.name.Substring(3)) + 1,
-            _lookAheadDistance,
-            pathPoints.Length - _lookAheadDistance - 1
-        );
+        if (foundIndex == (_pathPoints.Length - 2))
+        {
+            Finish();
+        }
+
+        if (foundIndex > PathFollower.PathPointIndex)
+        {
+            PathFollower.PathPointIndex = foundIndex;
+        }
 
         _lost = false;
+    }
+
+    /// <summary>
+    /// Calculates the potential energy for the enemy and applies the effects (charge, discharge, damage
+    /// </summary>
+    private void EnergyBehaviour()
+    {
+        // Calculate energy potential
+        _potentialEnergy = _potentialEnergyRange - Vector3.Distance(transform.position, _pathPoints[PathFollower.PathPointIndex].Position);
 
-        if (_waypointIndex == (pathPoints.Length - _lookAheadDistance - 1))
+        if (_potentialEnergy >= 0)
         {
-            Finish();
+            Charge(0.1f);
+        }
+        else
+        {
+            Discharge(0.1f);
+
+            if (_potentialEnergy < _potentialEnergyRange / 2)
+            {
+                _lost = true;
+            }
+        }
+
+        if (_energyCharge < 0)
+        {
+            Damage(30);
```

Comment:
This damage should be configurable

Reply:
this was a soft death taking a maximum of 4 ticks, replaced with Explode()